lib/jamaica/games/unit_game.dart

```diff
@@ -39,22 +39,38 @@ class UnitGame extends StatefulWidget {
 class _UnitGameState extends State<UnitGame> {
   List<_ChoiceDetail> choiceDetails = [];
   List<_ChoiceDetail> answerDetails = [];
-  Map<String, int> data = {};
   int score = 5;
   String displayText = '';
 
   @override
   void initState() {
     super.initState();
-    data['H'] = widget.question ~/ 100;
-    data['T'] = (widget.question ~/ 10) % 10;
-    data['U'] = widget.question % 10;
+    displayText = widget.question.toString();
+    int i = 0;
+    int j = displayText.length - 1;
+    String position = '';
+    while (i < 3) {
```

Comment:
i is always 0. how does this logic work?

Reply:
i am incrementing i down. Here i am trying to add digits in order starting from Unit place